lib/smart_answer_flows/calculate-your-holiday-entitlement.rb

```diff
@@ -124,22 +133,28 @@ def define
         from { Date.civil(1.year.ago.year, 1, 1) }
         to { Date.civil(1.year.since(Time.zone.today).year, 12, 31) }
 
-        calculate :leave_year_start_date do |response|
-          leave_year_start_date = response
-          if leaving_date.present?
-            raise InvalidResponse, :error_end_date_before_start_date if leaving_date <= leave_year_start_date
+        on_response do |response|
+          calculator.leave_year_start_date = response
+        end
 
-            raise InvalidResponse, :error_end_date_outside_leave_year_range unless YearRange.new(begins_on: leave_year_start_date).include?(leaving_date)
-          end
-          if start_date
-            raise InvalidResponse, :error_start_date_before_start_leave_year_date if start_date <= leave_year_start_date
-            raise InvalidResponse, :error_start_date_outside_leave_year_range unless YearRange.new(begins_on: leave_year_start_date).include?(start_date)
-          end
-          leave_year_start_date
+        validate :error_end_date_before_start_date do
+          calculator.leaving_date.blank? || calculator.leave_year_start_date.before?(calculator.leaving_date)
+        end
+
+        validate :error_end_date_outside_leave_year_range do
+          calculator.leaving_date.blank? || YearRange.new(begins_on: calculator.leave_year_start_date).include?(calculator.leaving_date)
+        end
+
+        validate :error_start_date_before_start_leave_year_date do
+          calculator.start_date.blank? || calculator.leave_year_start_date.before?(calculator.start_date)
+        end
+
+        validate :error_start_date_outside_leave_year_range do
+          calculator.start_date.blank? || YearRange.new(begins_on: calculator.leave_year_start_date).include?(calculator.start_date)
```

Comment:
Perhaps pedantic, but `if start_date` is different to `if start_date.present?`. As the first returns true if start_date is `" "` and the other false.

I think this should be:
```ruby
calculator.start_date.nil? || YearRange.new(begins_on: calculator.leave_year_start_date).include?(calculator.start_date)
```
Same for the `:error_start_date_before_start_leave_year_date` validate block.

The other validate blocks here are OK using `calculator.leaving_date.blank?` because the original was `if leaving_date.present?`

Reply:
Updated as per suggestion - thanks 👍